plugins/database/quota/src/org/apache/cloudstack/api/response/QuotaResponseBuilderImpl.java

```diff
@@ -0,0 +1,419 @@
+import org.apache.cloudstack.api.command.QuotaEmailTemplateUpdateCmd;
+import org.apache.cloudstack.api.command.QuotaStatementCmd;
+import org.apache.cloudstack.api.command.QuotaTariffListCmd;
+import org.apache.cloudstack.api.command.QuotaTariffUpdateCmd;
+import org.apache.cloudstack.quota.QuotaService;
+import org.apache.cloudstack.quota.constant.QuotaConfig;
+import org.apache.cloudstack.quota.constant.QuotaTypes;
+import org.apache.cloudstack.quota.dao.QuotaBalanceDao;
+import org.apache.cloudstack.quota.dao.QuotaCreditsDao;
+import org.apache.cloudstack.quota.dao.QuotaEmailTemplatesDao;
+import org.apache.cloudstack.quota.dao.QuotaTariffDao;
+import org.apache.cloudstack.quota.vo.QuotaBalanceVO;
+import org.apache.cloudstack.quota.vo.QuotaCreditsVO;
+import org.apache.cloudstack.quota.vo.QuotaEmailTemplatesVO;
+import org.apache.cloudstack.quota.vo.QuotaTariffVO;
+import org.apache.cloudstack.quota.vo.QuotaUsageVO;
+import org.apache.cloudstack.region.RegionManager;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import javax.ejb.Local;
+import javax.inject.Inject;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+@Component
+@Local(value = QuotaResponseBuilderImpl.class)
+public class QuotaResponseBuilderImpl implements QuotaResponseBuilder {
+    private static final Logger s_logger = Logger.getLogger(QuotaResponseBuilderImpl.class.getName());
+
+    @Inject
+    private QuotaTariffDao _quotaTariffDao;
+    @Inject
+    private QuotaBalanceDao _quotaBalanceDao;
+    @Inject
+    private QuotaCreditsDao _quotaCreditsDao;
+    @Inject
+    private QuotaEmailTemplatesDao _quotaEmailTemplateDao;
+
+    @Inject
+    private UserDao _userDao;
+    @Inject
+    private QuotaService _quotaService;
+    @Inject
+    AccountDao _accountDao;
+    @Inject
+    private RegionManager _regionMgr;
+
+    @Override
+    public QuotaTariffResponse createQuotaTariffResponse(QuotaTariffVO tariff) {
+        final QuotaTariffResponse response = new QuotaTariffResponse();
+        response.setUsageType(tariff.getUsageType());
+        response.setUsageName(tariff.getUsageName());
+        response.setUsageUnit(tariff.getUsageUnit());
+        response.setUsageDiscriminator(tariff.getUsageDiscriminator());
+        response.setTariffValue(tariff.getCurrencyValue());
+        response.setEffectiveOn(tariff.getEffectiveOn());
+        response.setDescription(tariff.getDescription());
+        response.setCurrency(QuotaConfig.QuotaCurrencySymbol.value());
+        return response;
+    }
+
+    @Override
+    public QuotaBalanceResponse createQuotaBalanceResponse(List<QuotaBalanceVO> quotaBalance, Date startDate, Date endDate) {
+        if (quotaBalance.size() == 0) {
```

Comment:
In order to avoid a ``NullPointerException``, this ``if`` condition should first check that ``quotaBalance`` is not null.  Also, consider using ``com.google.common.base.Preconditions#checkArgument`` for checking the validity of arguments.